classlib/src/main/java/org/teavm/classlib/java/util/function/TPredicate.java

```diff
@@ -36,4 +36,8 @@ default TPredicate<T> or(TPredicate<? super T> other) {
     default TPredicate<T> isEqual(Object targetRef) {
         return t -> TObjects.equals(t, targetRef);
     }
+
+    default <T> TPredicate<T> not(TPredicate<? super T> other) {
+        return (TPredicate<T>) other.negate();
```

Comment:
That's a good idea, but after correcting my slip-up of using `default` instead of `static` one can't do that anymore ;-). The problem is that the `test()` method is not static and therefor can't be called from the (correct) method `static <T> TPredicate<T> not(TPredicate<? super T> other)`.

Reply:
What about `other.test()`?